presto-orc/src/main/java/com/facebook/presto/orc/writer/SliceDictionaryColumnWriter.java

```diff
@@ -208,19 +230,20 @@ protected void writePresentAndDataStreams(
             PresentOutputStream presentStream,
             LongOutputStream dataStream)
     {
-        checkState(optionalSortedIndex.isPresent(), "originalDictionaryToSortedIndex is null");
-        int[] originalDictionaryToSortedIndex = optionalSortedIndex.get();
+        checkState(optionalSortedIndex.isPresent() == sortDictionaryKeys, "Sorted Index and Options are inconsistent %s", sortDictionaryKeys);
         for (int position = 0; position < rowGroupValueCount; position++) {
             presentStream.writeBoolean(rowGroupIndexes[position] != NULL_INDEX);
         }
+
+        int[] sortedIndexes = optionalSortedIndex.orElse(null);
         for (int position = 0; position < rowGroupValueCount; position++) {
-            int originalDictionaryIndex = rowGroupIndexes[position];
-            if (originalDictionaryIndex != NULL_INDEX) {
-                int sortedIndex = originalDictionaryToSortedIndex[originalDictionaryIndex];
-                if (sortedIndex < 0) {
-                    throw new IllegalArgumentException();
+            int dictionaryIndex = rowGroupIndexes[position];
+            if (dictionaryIndex != NULL_INDEX) {
+                int entryIndex = sortDictionaryKeys ? sortedIndexes[dictionaryIndex] : dictionaryIndex;
```

Comment:
A new branch is introduced here. But sortDictionarykeys will be used mostly at the environment level, so branch prediction will avoid the branching cost as it is either going to be true or false. The benchmark did not change prior and after this change. But if each writer can customize, the branch needs to be lifted out and code needs to be copied.

Reply:
After my map change got reverted due to the Performance, I decided not to take a chance and lifted the conditional operator outside of the loop.